src/main/resources/main.js

```diff
@@ -0,0 +1,178 @@
+const taskLib = require('/lib/xp/task');
+const eventLib = require('/lib/xp/event');
+const webSocketLib = require('/lib/xp/websocket');
+const appLib = require('/lib/xp/app');
+const store = require('/lib/store');
+
+const SOCKET_GROUP = 'welcome-app';
+exports.SOCKET_GROUP = SOCKET_GROUP;
+
+let listening = false;
+
+const bean = __.newBean('com.enonic.xp.app.welcome.WelcomePageScriptBean');
+
+/*    var xpEvent = {
+        "description": "Long running task",
+        "id": "7ca603c1-3b88-4009-8f30-46ddbcc4bb19",
+        "name": "task-7ca603c1-3b88-4009-8f30-46ddbcc4bb19",
+        "state": "RUNNING",
+        "application": "com.enonic.myapp",
+        "user": "user:store:me",
+        "startTime": "2017-10-01T09:00:00Z",
+        "progress": {
+            "info": "Processing item 33",
+            "current": 33,
+            "total": 42
+        }
+    };*/
+function handleXPTaskEvent(xpEvent) {
+    const task = xpEvent.data;
+
+    let cachedTask = store.getTask(task.id);
```

Comment:
lask_lib.get and lask_lib.list basically gives you the same functionality but without need to have your own `store` and locks. Moreover, it is unlikely that your code will work in GraalJS - because there will be multiple engines and your `store`  object will be different in every engine

Reply:
I started with lib-cache, but there was a problem with missing data. As far as I can tell it was caused by simultaneous modification of an entry.
I had to use single entry for all items because I had a need to search by 2 different fields which is impossible to do when you have variable cache keys.
I solved the issue with this implementation with access lock.

Reply:
We are starting to migrate to GraalJS. This code won't work as it assumes single engine